spanner/google/cloud/spanner/streamed.py

```diff
@@ -122,17 +130,49 @@ def _merge_values(self, values):
             field = self.fields[index]
             self._current_row.append(_parse_value_pb(value, field.type))
             if len(self._current_row) == width:
-                self._rows.append(self._current_row)
+                self._pending_rows.append(self._current_row)
                 self._current_row = []
 
+    def _flush_pending_rows(self):
+        """Helper for :meth:`consume_next`."""
+        flushed = self._pending_rows[:]
+        self._pending_rows[:] = ()
+        self._complete_rows.extend(flushed)
+
+    def _do_restart(self):
+        """Helper for :meth:`consume_next`."""
+        if not self._resume_token:
```

Comment:
_resume_token might be absent because the stream broke before yielding the first response or it yielded a few responses none of which had resume token. In either case you can just restart the stream from beginning instead of raising error.

Reply:
> or it yielded a few responses none of which had resume token

Seriously? Ouch.